spec/timezone.html

```diff
@@ -38,26 +31,23 @@ <h1>Time Zone Names</h1>
       <h1>
         IsValidTimeZoneName (
           _timeZone_: a String,
-        )
+        ): a Boolean
       </h1>
       <dl class="header">
         <dt>description</dt>
-        <dd>It returns *true* if _timeZone_ is an ASCII-case-insensitive match for a built-in time zone name and *false* otherwise.</dd>
+        <dd>The returned value is *true* if _timeZone_ is an ASCII-case-insensitive match for a built-in time zone name, and *false* otherwise.</dd>
       </dl>
-      <p>
-        An ECMAScript implementation that includes the ECMA-402 Internationalization API must implement the IsValidTimeZoneName abstract operation as specified in the ECMA-402 specification.
-      </p>
       <p>
         Once IsValidTimeZoneName(_timeZone_) has returned *true*, for the lifetime of the surrounding agent, IsValidTimeZoneName(_variant_) must return *true* if _variant_ is an ASCII-case-insensitive match for either _timeZone_ or CanonicalizeTimeZoneName(_timeZone_).
       </p>
       <p>
         For the purposes of this section, a String value _A_ is an ASCII-case-insensitive match for String value _B_ if the String value derived from _A_ by replacing each occurrence of a lowercase ASCII letter code unit (0x0061 through 0x007A, inclusive) with the corresponding uppercase ASCII letter code unit (0x0041 through 0x005A, inclusive) while preserving all other code units is exactly the same sequence of code units as the String value that is derived from _B_ in the same way.
       </p>
-      <p>The minimum implementation of IsValidTimeZoneName for ECMAScript implementations that do not include the ECMA-402 API, supporting only the *"UTC"* time zone, performs the following steps when called:
-      </p>
 
       <emu-alg>
-        1. If _timeZone_ is an ASCII-case-insensitive match for *"UTC"*, return *true*.
+        1. Let _timeZones_ be AvailableTimeZones().
```

Comment:
Since even the ECMA-262 AvailableTimeZones in this PR is constrained to the IANA Time Zone Database, doesn't this algorithm preclude any other purported time zone name (such as "SYSTEM" or even "-05:00") from being recognized as valid? In which case, isn't the preceding encouragement to use "Zone and Link names of the IANA Time Zone Database (and <strong>only</strong> such names)" actually a _requirement_, and therefore a new normative restriction in ECMA-262?

Reply:
"-05:00" is handled separately, strings that can be parsed by TimeZoneNumericUTCOffset are always "available". I guess allowing "SYSTEM" is incompatible with your first comment though, about collapsing the ECMA-402 AvailableTimeZones into the ECMA-262 one?

It seems like we could solve this by changing the sentence "The returned List is a sorted List of supported Zone and Link names in the IANA Time Zone Database." in the ECMA-262 AvailableTimeZones, to something like "The returned List is a sorted List of identifiers for time zones for which the implementation includes local political rules, and aliases for those identifiers."

Reply:
Yes, I believe that would avoid the normative change. But for the record, I think some kind of normative change _should_ be pursued—it makes me uncomfortable that behavior valid in an ECMA-262 implementation is disallowed by ECMA-402, making adoption of ECMA-402 a backwards-incompatible change for at least a hypothetical set of implementations. It seems to go against the spirit of ECMA-402 (which is generally quite liberal regarding implementation-defined behavior), and I'm not aware of any other aspect that is so constrained. Is there a ticket to bring them into accord, either by relaxing ECMA-402 or tightening ECMA-262?

Reply:
We have https://github.com/tc39/proposal-temporal/issues/1996, maybe that's what you're looking for?

Reply:
That looks related but different. It takes as given that "_implementations without Intl may choose to support non-IANA time zones, while 402-compliant implementations must not support any non-IANA time zones_", but I'm advocating a normative change to invalidate that statement.

Reply:
#2209